go/examples_test.go

```diff
@@ -99,3 +99,16 @@ func TestGetSecretWithGCPAuthIAM(t *testing.T) {
 		t.Fatalf("Expected %s, but got %s", expected, value)
 	}
 }
+
+func TestGetSecretWithKubernetesAuth(t *testing.T) {
+	if os.Getenv("LOCAL_TESTING") == "" {
```

Comment:
If you want to get this running in CI at any point, there is an example of getting a `kind` cluster running here: https://github.com/hashicorp/vault-csi-provider/blob/master/.circleci/config.yml#L73-L93.